src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2065,7 +2065,7 @@ private IOperation VisitUserDefinedBinaryConditionalOperator(IBinaryOperation bi
             SpillEvalStack();
             int captureId = captureIdForResult ?? _captureIdDispenser.GetNextId();
 
-            ref BasicBlockBuilder lazyFallThrough = ref stopSense ? ref fallToTrueOpt : ref fallToFalseOpt;
+            ref BasicBlockBuilder lazyFallThrough = ref stopValue ? ref fallToTrueOpt : ref fallToFalseOpt;
```

Comment:
@AlekseyTs this was the bug. Rather than stopSense (which is true in the case of something like `!(b || c)`), we need to be looking at the actual desired value at the end.